Modules/Scripted/ShapeAnalysisToolBox/DataImporter.py

```diff
@@ -904,18 +1233,404 @@ def importFiles(self, filePaths):
+    #pyQt4
+    except:
+      header = self.InputFreeSurferSegmentsTable.horizontalHeader()
+      header.setResizeMode(self.freesurferSubjectImport, qt.QHeaderView.ResizeToContents)  
+      header.setResizeMode(self.freesurferSubjectName, qt.QHeaderView.Stretch) 
+
+  def addRowToFreeSurferSegmentsTable(self,segment_name,label_id):
+    #create checkbox with a centered layout
+    check_box = qt.QCheckBox()
+    check_box.setChecked(False)
+    check_box.stateChanged.connect(lambda state, x=label_id: self.onToggleFreeSurferSegmentSelection(x))
+
+    container = qt.QWidget();
+    layout = qt.QHBoxLayout(container);
+    layout.addWidget(check_box);
+    layout.setAlignment(qt.Qt.AlignCenter);
+    layout.setContentsMargins(0,0,0,0);
+    container.setLayout(layout);
+
+    rowPosition = self.InputFreeSurferSegmentsTable.rowCount
+    self.InputFreeSurferSegmentsTable.insertRow(rowPosition)
+    self.InputFreeSurferSegmentsTable.setCellWidget(rowPosition , self.freesurferSegmentsImport,container)
+    self.InputFreeSurferSegmentsTable.setItem(rowPosition , self.freesurferSegmentsName, qt.QTableWidgetItem(segment_name))
+
+  def uncheckFreeSurferTables(self):
+    #uncheck subjects
+    subject_number = self.InputFreeSurferSubjectsTable.rowCount
+    for i_row in range(subject_number):
+      self.InputFreeSurferSubjectsTable.cellWidget(i_row,0).children()[1].blockSignals(True)
+      self.InputFreeSurferSubjectsTable.cellWidget(i_row,0).children()[1].setChecked(False)
+      self.InputFreeSurferSubjectsTable.cellWidget(i_row,0).children()[1].blockSignals(False)
+
+    #uncheck segments
+    segment_number = self.InputFreeSurferSegmentsTable.rowCount
+    for i_row in range(segment_number):
+      self.InputFreeSurferSegmentsTable.cellWidget(i_row,0).children()[1].blockSignals(True)
+      self.InputFreeSurferSegmentsTable.cellWidget(i_row,0).children()[1].setChecked(False)
+      self.InputFreeSurferSegmentsTable.cellWidget(i_row,0).children()[1].blockSignals(False)
+  
+  def resetFreeSurferTab(self):
+    self.freesurfer_wanted_segments=[]
+    self.uncheckFreeSurferTables()
+  
+  def resetCSVTab(self):
+    #reset CSV tab
+    self.InputCSVFileNameLineEdit.text=''
+
+  def resetDirectoryTab(self):
+    #reset directroy tab
+    self.InputFolderNameLineEdit.text=''
+    self.FolderDirectoryButton.blockSignals(True)
+    self.FolderDirectoryButton.directory = '/'
+    self.FolderDirectoryButton.blockSignals(False)
   '''
   GUI Callback functions
   '''
+  #
+  #Shape Analysis Structure Generation Callbacks
+  #
+  def onShapeAnalysisFolderChanged(self,directoryPath):
+    logging.debug("onShapeAnalysisFolderChanged: {}".format(directoryPath))
+    # Create a list of files from the directory
+    directory = qt.QDir(directoryPath)
+    if not directory.exists():
+      logging.error("Directory {} does not exist.".format(directory))
+      return
+
+    self.shape_analysis_folder = directoryPath
```

Comment:
Remove ivar `shape_analysis_folder`, value can directly be retrieved from `self.InputShapeAnalysisFolderNameLineEdit.text`

To avoid having to explicitly reference the widget, you could also add a property like this:

```
  @property
  def inputShapeAnalysisPath(self):
    return self.InputShapeAnalysisFolderNameLineEdit.text
```

Then, you could use `self.inputShapeAnalysisPath` to get the value